qa/rpc-tests/z_sendmany.py

```diff
@@ -0,0 +1,144 @@
+        txid = wait_and_assert_operationid_status(self.nodes[1], myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        hex = self.nodes[1].gettransaction(txid)['hex']
+        vouts = self.nodes[1].decoderawtransaction(hex)['vout']
+        assert_equal(len(vouts),1)
+        assert_equal(vouts[0]['scriptPubKey']['addresses'][0],fromAddress)
+        assert_equal(self.nodes[2].z_getbalance(ZDestAddress),10)
+
+
+        #Send 10 Zen to Taddress nd verify the change is returned back to the input address of the transaction
+        TDestAddress = self.nodes[2].getnewaddress()
+        recipients= [{"address":TDestAddress, "amount": amount}]
+        myopid = self.nodes[1].z_sendmany(fromAddress,recipients,1,self.FEE, True)
+        txid = wait_and_assert_operationid_status(self.nodes[1], myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        hex = self.nodes[1].gettransaction(txid)['hex']
+        vouts = self.nodes[1].decoderawtransaction(hex)['vout']
+        assert_equal(len(vouts),2)
+        oldAddress = True
+        for vout in vouts:
+            if (vout['scriptPubKey']['addresses'][0] != TDestAddress and vout['scriptPubKey']['addresses'][0] != fromAddress):
+                oldAddress = False
+        assert_equal(oldAddress,True)
+        assert_equal(self.nodes[2].getbalance(),amount)
+
+
+        #Send 10 Zen to Taddress nd verify the change is sent to a new address
+        recipients= [{"address":TDestAddress, "amount": amount}]
+        myopid = self.nodes[1].z_sendmany(fromAddress,recipients,1,self.FEE, False)
+        txid = wait_and_assert_operationid_status(self.nodes[1], myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        hex = self.nodes[1].gettransaction(txid)['hex']
+        vouts = self.nodes[1].decoderawtransaction(hex)['vout']
+        assert_equal(len(vouts),2)
+        newAddress = True
+        for vout in vouts:
+            if (vout['scriptPubKey']['addresses'][0] == fromAddress):
+                newAddress = False
+        assert_equal(newAddress,True)
+        assert_equal(self.nodes[2].getbalance(),amount*2)
+
+
+        #Send 10 Zen to Taddress nd verify the change is sent to a new address
+        recipients= [{"address":fromAddress, "amount": amount}]
+        myopid = self.nodes[2].z_sendmany(TDestAddress,recipients)
+        txid = wait_and_assert_operationid_status(self.nodes[2], myopid)
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        hex = self.nodes[2].gettransaction(txid)['hex']
+        vouts = self.nodes[2].decoderawtransaction(hex)['vout']
+        assert_equal(len(vouts),2)
+        newAddress = True
+        for vout in vouts:
+            if (vout['scriptPubKey']['addresses'][0] == TDestAddress):
+                newAddress = False
+        assert_equal(newAddress,True)
+
```

Comment:
An additional test case returning the change to fromaddress where fromaddress is a multisig address would be good (I've verified that it does work with multisig).